Build failure with libarrow 20.0

<!--
First look if there is already a similar bug report. If there is, upvote the issue with 👍
-->

**Describe the bug**
librerun is failing to build when compiled against libarrow 20.0, which is discovered by https://github.com/conda-forge/librerun-sdk-feedstock/pull/39. The failure appears to be caused by librerun attempting to access a class member that has been made protected in the libarrow 20.0 update ([comment](https://github.com/conda-forge/librerun-sdk-feedstock/pull/39#issuecomment-2844147749) thanks to @h-vetinari):

GCC:
```
$SRC_DIR/rerun_cpp/src/rerun/time_column.cpp:23:56:   required from here
$BUILD_PREFIX/lib/gcc/x86_64-conda-linux-gnu/13.3.0/include/c++/bits/stl_construct.h:119:7: error: 'arrow::PrimitiveArray::PrimitiveArray(const std::shared_ptr<arrow::DataType>&, int64_t, const std::shared_ptr<arrow::Buffer>&, const std::shared_ptr<arrow::Buffer>&, int64_t, int64_t)' is protected within this context
  119 |       ::new((void*)__p) _Tp(std::forward<_Args>(__args)...);
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from $SRC_DIR/rerun_cpp/src/rerun/time_column.cpp:6:
$PREFIX/include/arrow/array/array_base.h:284:3: note: declared protected here
  284 |   PrimitiveArray(const std::shared_ptr<DataType>& type, int64_t length,
      |   ^~~~~~~~~~~~~~
```

MSVC:
```
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.29.30133\include\xutility(142): error C2248: 'arrow::PrimitiveArray::PrimitiveArray': cannot access protected member declared in class 'arrow::PrimitiveArray'
%PREFIX%\Library\include\arrow/array/array_base.h(284): note: see declaration of 'arrow::PrimitiveArray::PrimitiveArray'
%PREFIX%\Library\include\arrow/array/array_base.h(278): note: see declaration of 'arrow::PrimitiveArray'
```

**To Reproduce**
Steps to reproduce the behavior:
1. <!-- Go to '…' -->
2. <!-- Click on '…' -->
3. <!-- Scroll down to '…' -->
4. <!-- See error -->

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->

**Screenshots**
<!-- If applicable, add screenshots to help explain your problem. -->

**Backtrace**
<!-- If this is a crash, include the backtrace in a code-block (```[STACKTRACE]```). -->

**Desktop (please complete the following information):**
 - OS: <!-- e.g. macOS Monterey 12.6 -->

**Rerun version**
<!-- Paste the output of `rerun --version` here. -->

**Additional context**
<!-- Add any other context about the problem here. -->

--- a/rerun_cpp/src/rerun/time_column.cpp
+++ b/rerun_cpp/src/rerun/time_column.cpp
@@ -4,6 +4,7 @@
 #include "c/rerun.h"
 
 #include <arrow/array/array_base.h>
+#include <arrow/array/util.h>
 #include <arrow/buffer.h>
 #include <arrow/c/bridge.h>
 
@@ -20,7 +21,9 @@ namespace rerun {
         // which may or may not be another copy (if the collection already owns the data this is just a move).
         auto length = static_cast<int64_t>(times.size());
         auto buffer = arrow_buffer_from_vector(std::move(times).to_vector());
-        array = std::make_shared<arrow::PrimitiveArray>(datatype, length, buffer);
+        auto buffers = std::vector<std::shared_ptr<arrow::Buffer>>{nullptr, buffer};
+        auto array_data = std::make_shared<arrow::ArrayData>(datatype, length, std::move(buffers));
+        array = arrow::MakeArray(array_data);
     }
 
     TimeColumn TimeColumn::from_seconds(

